modules/imgcodecs/src/grfmt_gif.cpp

```diff
@@ -436,11 +437,18 @@ bool GifDecoder::getFrameCount_() {
             // Application Extension need to be handled for the loop count
             if (extension == 0xFF) {
                 int len = m_strm.getByte();
+                bool isFoundNetscape = false;
                 while (len) {
-                    // TODO: In strictly, Application Identifier and Authentication Code should be checked.
-                    if (len == 3) {
+                    if (len == 11) {
+                        String app_auth_code(len, ' ');
+                        m_strm.getBytes(const_cast<void*>(static_cast<const void*>(app_auth_code.data())), len);
+                        isFoundNetscape = (memcmp(app_auth_code.data(), "NETSCAPE2.0", len) == 0);
+                    }  else if ( (len == 3) && (isFoundNetscape) ) {
```

Comment:
Shouldn't there be an interpretation if `!isFoundNetscape` too ?

Reply:
Thank you for comment, I fixed it to show log message when it is unexpected condition.